out_writeapi.go

```diff
@@ -225,19 +258,42 @@ func sendRequestExactlyOnce(ctx context.Context, data [][]byte, config **outputC
 	return nil
 }
 
+// this function enables synchronous retries and rebuilding a valid stream based on the server response
 func sendRequestRetries(ctx context.Context, data [][]byte, config **outputConfig) error {
 	retryer := newStatelessRetryer((*config).numRetries)
 	attempt := 0
+	//increment describes whether to increase the attempt count
+	var increment int
+	//backoffPeriod describes the time to wait between retry attempts
+	var backoffPeriod time.Duration
```

Comment:
Instead of introducing these 2 extra variables "increment" and "backoffPeriod", why not simply update "attempt" and perform "sleep" only in the "else" clause below?

Reply:
Done.